openmcyclus/depletion.py

```diff
@@ -3,156 +3,192 @@
-        by cyclus.
+        Creates a list of each of the spent fuel compositions from the 
+        OpenMC depletion
 
         Parameters:
         -----------
+        material_id: list of strs
+            material ids for the assembly materials in the OpenMC model
+        path: str
+            path of directory holding the files for/from OpenMC
 
-        Outputs:
+        Returns:
         --------
-        {self.path}/{self.prototype}_fuel.xml : XML file
-            File containing the depleted material composition in the
-            required format for Cyclus. The path and prototype names are read
-            in from the class instantiation. Path will need to
-            be relative to Cyclus input file location
+        spent_comps: list of dicts
+            list of the compositions from the OpenMC model
         '''
-        results = od.Results(self.path / "depletion_results.h5")
-        composition = results.export_to_materials(-1)
-        root = minidom.Document()
-        recipe = root.createElement('recipes')
-        root.appendChild(recipe)
-        for index, material in enumerate(composition):
-            material_recipe = root.createElement('recipe')
-            name = root.createElement('name')
-            basis = root.createElement('basis')
-
-            name_text = root.createTextNode(material.name)
-            basis_text = root.createTextNode('atom')
-
-            name.appendChild(name_text)
-            basis.appendChild(basis_text)
-            material_recipe.appendChild(name)
-            material_recipe.appendChild(basis)
-            for item in composition[index].nuclides:
-                nuclide = root.createElement('nuclide')
-                nuclide_id = root.createElement('id')
-                nuclide_comp = root.createElement('comp')
-
-                id_text = root.createTextNode(item.name)
-                comp_text = root.createTextNode(str(item.percent))
-
-                nuclide_id.appendChild(id_text)
-                nuclide_comp.appendChild(comp_text)
-                nuclide.appendChild(nuclide_id)
-                nuclide.appendChild(nuclide_comp)
-                material_recipe.appendChild(nuclide)
-            recipe.appendChild(material_recipe)
-        xml_str = root.toprettyxml(newl='\n')
-        file_name = str(self.path / str(self.prototype + "_fuel.xml"))
-        with open(file_name, "w") as f:
-            f.write(xml_str)
-        return
+        results = od.Results(path + "depletion_results.h5")
+        spent_comps = []
+        for index, material_id in enumerate(material_ids):
+            material = results[-1].get_material(material_id)
+            comp = {}
+            for nuclide in material.nuclides:
+                if nuclide.percent < 1e-15:
+                    continue
+                Z, A, m = openmc.data.zam(nuclide.name)
+                comp.update({Z*10000000+A*10000 + m :nuclide.percent/100})
```

Comment:
Also as a style note, it's good to be consistent with using scientific notation vs not (when considering magnitudes).